Dockerfile

```diff
@@ -16,7 +14,7 @@ COPY controllers/ controllers/
 COPY pkg/ pkg/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o manager main.go
```

Comment:
Was there a reason for the `-a` being in here?

Reply:
Not sure, but this would cause a rebuild everytime. causing the docker layer caching to never work.

In follow up PRs we use --cache with buildx that uses a go build cache to build binaries for both architectures. Using that build cache we can speed up builds a lot.

Reply:
The makefile and Dockerfile were generated by kubebuilder so there are some opinionated things in there.